tests/ignite/engine/test_custom_events.py

```diff
@@ -296,6 +313,79 @@ def test_every_event_filter_with_engine():
     _test_every_event_filter_with_engine()
 
 
+def test_before_event_filter_with_engine():
+
+    data = range(100)
+
+    def _test(event_name, event_attr, before, expect_calls):
+        engine = Engine(lambda e, b: 1)
+        num_calls = 0
+
+        @engine.on(event_name(before=before))
+        def _before_event():
+            nonlocal num_calls
+            num_calls += 1
+            assert getattr(engine.state, event_attr) < before
+
+        engine.run(data, max_epochs=5)
+        assert num_calls == expect_calls
+
+    _test(Events.ITERATION_COMPLETED, "iteration", 0, 0)
+    _test(Events.ITERATION_COMPLETED, "iteration", 300, 299)
+    _test(Events.ITERATION_COMPLETED, "iteration", 501, 500)
+
+    _test(Events.EPOCH_COMPLETED, "epoch", 0, 0)
+    _test(Events.EPOCH_COMPLETED, "epoch", 3, 2)
+    _test(Events.EPOCH_COMPLETED, "epoch", 6, 5)
+
+
+def test_after_event_filter_with_engine():
+
+    data = range(100)
+
+    def _test(event_name, event_attr, after, expect_calls):
+        engine = Engine(lambda e, b: 1)
+        num_calls = 0
+
+        @engine.on(event_name(after=after))
+        def _after_event():
+            nonlocal num_calls
+            num_calls += 1
+            assert getattr(engine.state, event_attr) > after
+
+        engine.run(data, max_epochs=5)
+        assert num_calls == expect_calls
+
+    _test(Events.ITERATION_STARTED, "iteration", 0, 500)
+    _test(Events.ITERATION_COMPLETED, "iteration", 300, 200)
+    _test(Events.ITERATION_COMPLETED, "iteration", 500, 0)
+
+    _test(Events.EPOCH_STARTED, "epoch", 0, 5)
+    _test(Events.EPOCH_COMPLETED, "epoch", 3, 2)
+    _test(Events.EPOCH_COMPLETED, "epoch", 5, 0)
+
+
+def test_before_and_after_event_filter_with_engine():
+
+    data = range(100)
+
+    def _test(event_name, event_attr, before, after, expect_calls):
+        engine = Engine(lambda e, b: 1)
+        num_calls = 0
+
+        @engine.on(event_name(before=before, after=after))
+        def _before_and_after_event():
+            nonlocal num_calls
+            num_calls += 1
+            assert getattr(engine.state, event_attr) > after
+
+        engine.run(data, max_epochs=5)
+        assert num_calls == expect_calls
+
+    _test(Events.ITERATION_STARTED, "iteration", 300, 100, 199)
+    _test(Events.EPOCH_COMPLETED, "epoch", 4, 1, 2)
```

Comment:
Thanks @louis-she , You can improve this test by using PyTest parameterization feature as follows:
```suggestion
@pytest.mark.parametrize(
    "event_name, before, after, expected_calls",
    [
        (Events.ITERATION_STARTED, 300, 100, 199),
        (Events.EPOCH_COMPLETED, 4, 1, 2)
    ],
)
def test_before_and_after_event_filter_with_engine(event_name, before, after, expected_calls):

    data = range(100)
    engine = Engine(lambda e, b: 1)
    num_calls = 0

   @engine.on(event_name(before=before, after=after))
   def _before_and_after_event():
        nonlocal num_calls
        num_calls += 1
        assert engine.state.get_event_attrib_value(event_name) > after

    engine.run(data, max_epochs=5)
    assert num_calls == expected_calls
```